spec/rspec/matchers_spec.rb

```diff
@@ -24,6 +24,16 @@
       end
     end
   end
+
+  describe "#respond_to?" do
+    it "returns true for methods in example scope" do
+      expect(self.respond_to?(:be_happy)).to eq(true)
+    end
+
+    it "returns false for methods not defined in matcher or example scope" do
+      expect(self.respond_to?(:have_eyes_closed)).to eq(true)
+    end
+  end
```

Comment:
There are a couple pitfalls with `respond_to?` that are easy to fall into that I think we should have specs for:

* The 2nd arg (`include_private`) is easy to forget to support.  Would be nice to have a spec like:

``` ruby
it "supports the optional `include_private` arg" do
  expect(self.respond_to?(:puts, true)).to eq true
  expect(self.respond_to?(:puts, false)).to eq false
  expect(self.respond_to?(:puts)).to eq false
end
```

* If you define `respond_to?` on 1.9 instead of `respond_to_missing?`, the respond to checks you've done here would all pass.  The benefit of `respond_to_missing?` over `respond_to?` is it allows `method` to work, so a spec that requires that `respond_to_missing?` is used would be nice:

``` ruby
it "allows `method` to get dynamic matcher methods", :if => RUBY_VERSION.to_f >= 1.9 do
  expect(method(:be_happy).call).to be_a(be_happy.class)
end
```

(Of course, it's good to confirm that these specs fail as expected when you break the implementation).